rusoto/core/src/reactor.rs

```diff
@@ -72,7 +73,7 @@ impl Reactor {
 
 /// A request dispatcher backed by an implicit event loop.
 pub struct RequestDispatcher {
-    sender: mpsc::UnboundedSender<(SignedRequest, oneshot::Sender<Result<HttpResponse, HttpDispatchError>>)>
+    sender: mpsc::UnboundedSender<((SignedRequest, Option<Duration>), oneshot::Sender<Result<HttpResponse, HttpDispatchError>>)>
```

Comment:
I've been wondering here whether the timeout `Duration` should be part of the `SignedRequest` for convenience, but then again it's probably a good idea to keep the concerns apart.

Reply:
I agree that it's a good idea to keep the concerns apart.